pkg/webhook/webhook.go

```diff
@@ -447,25 +474,23 @@ func MutateHandler(w http.ResponseWriter, req *http.Request) {
 		handleValidationError(w, ar, err)
 		return
 	}
-	if netSelections, exists := pod.ObjectMeta.Annotations[networksAnnotationKey]; exists && netSelections != "" {
+
+	defaultNetSelection, defExist := pod.ObjectMeta.Annotations[defaultNetworkAnnotationKey]
+	additionalNetSelections, addExists := pod.ObjectMeta.Annotations[networksAnnotationKey]
+
+	if defExist || addExists {
 		/* map of resources request needed by a pod and a number of them */
 		resourceRequests := make(map[string]int64)
 
-		/* unmarshal list of network selection objects */
-		networks, err := parsePodNetworkSelections(netSelections, pod.ObjectMeta.Namespace)
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
-
-		for _, n := range networks {
-			/* for each network in annotation ask API server for network-attachment-definition */
-			networkAttachmentDefinition, err := getNetworkAttachmentDefinition(n.Namespace, n.Name)
+		if defaultNetSelection != "" {
+			defNetwork, err := parsePodNetworkSelections(defaultNetSelection, pod.ObjectMeta.Namespace)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+			resourceRequests, err = parseNetSelectionResourceRequest(defNetwork[0], resourceRequests)
```

Comment:
Good finding on the defaultNetwork index issue!

About parsePodNetworkSelections, if it fails to unmarshall, chance is the defaultNetwork config is not a json format, but a string like `default/sriov-net` or any other malformat string. For the former, it returns a default network element, for the latter, it will fail in parsePodNetworkSelectionElement which results in err return in parsePodNetworkSelections. so we will either get a right network element or return error in parsePodNetworkSelections. did I miss something here?

what about I change the `defNetwork[0]` to a for loop against `defNetwork`? In that case, if defNetwork is empty, it will skip the loop.

Reply:
you're right. 

I think a list length check is sufficient.

Reply:
Done